src/cli/shell_hook.rs

```diff
@@ -148,7 +151,32 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         }
         // Skipping the activated environment caching for the script.
         // As it can still run scripts.
-        false => generate_activation_script(args.shell, &environment).await?,
+        false => {
+            let shell = get_shell(args.shell);
+            let script = generate_activation_script(shell.clone(), &environment).await?;
+            let in_shell = match std::env::var("PIXI_IN_SHELL") {
+                Ok(val) => val != "0" && !val.is_empty(),
+                Err(_) => false,
+            };
+            if project.config().change_ps1() && !in_shell {
+                let prompt_name = match environment.name() {
+                    EnvironmentName::Default => project.name().to_string(),
+                    EnvironmentName::Named(name) => format!("{}:{}", project.name(), name),
+                };
+                let prompt = match shell {
+                    ShellEnum::NuShell(_) => prompt::get_nu_prompt(prompt_name.as_str()),
+                    ShellEnum::PowerShell(_) => prompt::get_powershell_prompt(prompt_name.as_str()),
+                    ShellEnum::Bash(_) => prompt::get_posix_prompt(prompt_name.as_str()),
+                    ShellEnum::Zsh(_) => prompt::get_posix_prompt(prompt_name.as_str()),
+                    ShellEnum::Fish(_) => prompt::get_fish_prompt(prompt_name.as_str()),
+                    ShellEnum::Xonsh(_) => prompt::get_xonsh_prompt(),
+                    ShellEnum::CmdExe(_) => prompt::get_cmd_prompt(prompt_name.as_str()),
+                };
+                format!("{}\n{}", script, prompt)
+            } else {
+                script
+            }
```

Comment:
Let's move that logic into `generate_activation_script`

Reply:
That will require passing a few extra arguments that were scope in this function. Is that ok?

Reply:
Also this block is taken essentially from `shell.rs`... there's a bit of duplication.

Reply:
> That will require passing a few extra arguments that were scope in this function. Is that ok?

That's okay


> Also this block is taken essentially from shell.rs... there's a bit of duplication.

Would be great if you could extract that into a function then 🙂

Reply:
Any suggestions for the potential place? I looked at `/prompt.rs` initially but it doesn't have any imports, and the best next place I can think of is all they back in `rattler_shell` 😬 Is there a "shell utils" module somewhere?

Reply:
`prompt.rs` sounds good to me. What do you mean with "it doesn't have any imports"?

Reply:
https://github.com/prefix-dev/pixi/blob/main/src/prompt.rs looks like a collection of pure functions with no dependencies on external code, only the stdlib. Is it ok to import other stuff there for the types and so on?